### DISCUSSION TITLE
Error when using Tag.Structure

### DISCUSSION THREAD

**Discussion by MrlnHi** [Minestom/Minestom#906] [Help] [answered]
When I use 
```java
Tag.Structure("teams", GameTeam.class);
```
with GameTeam being a record that has a NamedTextColor as one of its fields, I get this exception:
```
Caused by: java.lang.IllegalArgumentException: Unsupported type: class net.kyori.adventure.text.format.NamedTextColor
```
how can I make it so it can get serialized correctly?

**Comment by TheMode** [comment, marked as answer]
Only a few types are valid inside records for auto-structure. They are listed [here](https://github.com/Minestom/Minestom/blob/1bb0af7f77064739c19ac18ef466f67721bc6c51/src/main/java/net/minestom/server/tag/TagRecord.java#L19).

The list may expand in the future, but for now you will have to create your own `TagSerializer`

**Reply by MrlnHi** [reply]
So there is no way of expanding that myself?

**Reply by TheMode** [reply]
Indeed.
